.storybook/decorators/withLifecycle.ts

```diff
@@ -0,0 +1,54 @@
+const setDisabledState = (element: HTMLButtonElement, state: boolean) => {
+  element.disabled = state;
+  element.classList.toggle('ais-ClearRefinements-button--disabled');
+};
+
+export const withLifecycle = (
```

Comment:
Any reasons to not go with a similar signature as `withHits` (with a single object)?

Reply:
The main reason is readability. The `withHits` decorator only takes a function as an argument but it's not the case with this one. It feels like with the list of parameters reads better, but we can change it I don't have a strong opinion.

```js
// with list of parameters
withLifecycle(search, container, () =>
  instantsearch.widgets.configure({
    hitsPerPage: 1
  })
);

// with named parameters
withLifecycle({
  search,
  container,
  fn(node) {
    instantsearch.widgets.searchBox({
      container: node
    });
  }
});
```

Reply:
For me, it has to be either `exactly same` or `don't care`. If we will introduce more decorators, then they'd better look the same, so that we can do

```js
withHits(...
  withLifecycle(...
    withFoo(...
      withBar(...
```

It will look cleaner. And ultimately we could do like this:

```js
withDecorators([withHits, withLifecycle, ...], ({ container, instantsearch, search }) => {
  ...
});
```

but if we're not going that far, I'm okay with the current signature.